pages/docs/tooling/tooling-performance-analysis.md

```diff
@@ -1,70 +1,283 @@
+You can then use the resulting file to analyse the profiling data.
 
-For a participant called `MySolver`, the files are called as follows:
+## Fundamental Events
 
-* `precice-MySolver-events.json`
-* `precice-MySolver-events-summary.log`
+Some events are useful for everyone, while others are only useful for developers.
+To keep this feature as useful as possible, we pre-selected a set of events, which we deem fundamental.
+These include the main preCICE API functions and the time spend between these calls inside the solvers.
+Fundamental events should give you an insight in the overhead of preCICE as well as load imbalance between ranks etc.
 
-## The events summary file
+Fundamental events are:
 
-The events summary file contains a table of events, their occurrences and some statistics on their runtime.
-This can be helpful to quickly identify where the preCICE library spends most of its time.
+* `SolverInterface` construction. This includes configuration and setting up the Intra-communication of each participant.
+* `solver.initialize` time spend in the solver until `initialize()` is called. This includes setting meshes, defining initial data and preparing the solver.
+* `initialize()` time spend in preCICE `initialize()`. This includes establishing communication between participants, mesh and data transfer, as well as mapping computation.
+* `solver.advance` time spend in the solver between `advance()` calls, including the time between `initialize()` and the first `advance()` call.
+* `advance()` time spend in preCICE `advance()`. This includes data mapping, data transfer, acceleration.
 
-It is especially helpful to focus on [noteworthy events](#noteworthy-events).
+## Measuring Blocking Operations
 
-This is an example output:
+Some parts of preCICE involve communication, which cannot be interleaved efficiently with other computations.
+Measuring the runtime of such operations can be tricky, as the time spend waiting should be ignored.
+Synchronizing all ranks using a barrier solved the issue, but has an impact on performance.
 
-```text
-Run finished at Wed Aug  1 09:41:10 2018
-Global runtime       = 12859ms / 12s
-Number of processors = 4
-# Rank: 0
+To keep this waiting component of the overall measurement to a minimum without affecting performance, we added a configuration option to toggle the synchonization before required events.
+Use the `sync-mode` attribute to enable such synchonization if you need it.
 
-  Event |      Count |  Total[ms] |    Max[ms] |    Min[ms] |    Avg[ms] |       T[%] |
----------------------------------------------------------------------------------------
-_GLOBAL |          1 |      12859 |      12859 |      12859 |      12859 |         99 |
-advance |          1 |         84 |         84 |         84 |         84 |          0 |
+```xml
+<precice-configuration sync-mode="1">
+    ...
+</precice-configuration>
+```
```

Comment:
This, we still want to change. Then, it could make sense to first introduce the profiling tag.